test/application_system_test_case.rb

```diff
@@ -4,16 +4,52 @@
 require "capybara/rails"
 require "capybara/minitest"
 
+require "axe/matchers/be_axe_clean"
+require "axe/expectation"
+
 require "test_helpers/cuprite_setup"
 
 Ferrum::Browser.new(process_timeout: 60, timeout: 60)
 
 class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
   driven_by :cuprite, using: :chrome, screen_size: [1400, 1400]
 
+  AXE_RULES_TO_SKIP = [:region].freeze
+  AXE_WITHIN_SELECTOR = "body"
+
   def with_preview(preview_name)
     component_uri = self.class.name.gsub("Test", "").gsub("Integration", "").underscore
 
     visit("/rails/view_components/primer/#{component_uri}/#{preview_name}")
+
+    assert_accessible(page)
+  end
+
+  def assert_accessible(page, within: AXE_WITHIN_SELECTOR, skipping: AXE_RULES_TO_SKIP, **options)
+    options[:within] = within
+    options[:skipping] = skipping
+
+    is_axe_clean = Axe::Matchers::BeAxeClean.new.tap do |a|
+      options.each do |option|
+        key, value = option
+        a.send(key, *value)
+      end
+    end
+
+    Axe::AccessibleExpectation.new.assert page, is_axe_clean
+  end
+
+  # Capybara Overrides
+  # From axe-core docs- "insert calls at each point in your tests where a new piece of UI becomes visible or exposed"
```

Comment:
Are `fill_in` and `click_button` the only ones that change the UI? Do we need these for other APIs as well?

Reply:
those are the only ones we use right now

Reply:
Can we implement the other APIs? For example, it's not clear that if I'm writing a system test and start using a new API that my component can become inaccessible without the tests catching it.

Reply:
Yes! I'll do this in a followup :)